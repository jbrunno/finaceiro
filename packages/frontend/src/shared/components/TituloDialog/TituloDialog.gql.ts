import { gql } from '@apollo/client';
import { TituloQuestionario } from '../TituloCard/TituloCard.gql';

export const TITULO_INSERT_MUTATION = gql`
  mutation TituloPost($input: TituloPostInput!) {
    tituloPost(input: $input) {
      id
    }
  }
`;

export const TITULO_EDIT_MUTATION = gql`
  mutation TituloPatch($input: TituloPatchInput!) {
    tituloPatch(input: $input) {
      id
    }
  }
`;

export type TituloInsertData = TituloId;

export type TituloInsertInput = TituloQuestionarioDialog;

export type TituloEditData = TituloId;

export type TituloEditInput = TituloQuestionarioDialog;

export type TituloQuestionarioDialog = TituloQuestionario & {
  id: string | null;
};

type TituloId = {
  id: string;
};
