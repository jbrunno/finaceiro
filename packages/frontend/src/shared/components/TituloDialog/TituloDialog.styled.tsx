import { styled, css } from '@frontend/styles';
import { RadioGroup, TextField } from '@frontend/components';

export const HeaderSyled = styled('div')(
  ({ theme }) => css`
    display: flex;
    align-items: right;
    gap: ${theme.fn.spacing(8)};
    margin: 0 auto;
    margin-bottom: ${theme.fn.spacing(16)};
  `,
);

export const FormStyled = styled('form')(
  ({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.fn.spacing(16)};
    min-width: ${theme.fn.size(600)};
  `,
);

export const GridSyled = styled('div')`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.fn.spacing(10)};
`;

export const TextFieldStyled = styled(TextField)`
  width: 100%;
  min-height: ${({ theme }) => theme.fn.size(60)};
`;

export const RadioButtonSyled = styled('div')`
  width: 100%;
`;

export const RadioGroupStyled = styled(RadioGroup)(
  ({ theme }) => css`
    display: flex;
    flex-direction: row;
    gap: ${theme.fn.spacing(8)};
    margin-bottom: ${theme.fn.spacing(12)};
  `,
);

export const ButtonGridSyled = styled('div')`
  display: flex;
  justify-content: center;
  margin-top: ${({ theme }) => theme.fn.spacing(24)};
`;
