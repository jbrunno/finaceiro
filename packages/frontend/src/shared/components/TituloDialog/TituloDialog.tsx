import {
  Dialog,
  FormControlLabel,
  Radio,
  Button,
  Tooltip,
  Typography,
  DialogContent,
} from '@frontend/components';
import { HelpOutline as HelpOutlineIcon } from '@frontend/icons';
import {
  FormFields,
  UseTituloDialogProps,
  useTituloDialog,
} from './hooks/useTituloDialog';
import {
  FormStyled,
  GridSyled,
  RadioButtonSyled,
  RadioGroupStyled,
  TextFieldStyled,
  ButtonGridSyled,
} from './TituloDialog.styled';

type TituloDialogProps = UseTituloDialogProps & {
  open: boolean;
  onClose: () => void;
  onSaved: () => void;
};

export function TituloDialog({
  open,
  onClose,
  onSaved,
  tituloId,
  tituloQuestionario,
  ...dialogProps
}: TituloDialogProps) {
  const { formik } = useTituloDialog({ onSaved, tituloId, tituloQuestionario });
  return (
    <Dialog
      {...dialogProps}
      maxWidth="xl"
      open={open}
      onClose={onClose}
      title="Informações do título"
    >
      <DialogContent>
        <FormStyled onSubmit={formik.handleSubmit}>
          <GridSyled>
            <TextFieldStyled
              variant="outlined"
              size="medium"
              label="Em qual site da FTX ocorreu o registro?"
              value={formik.values.siteRegistro}
              onBlur={formik.handleBlur(FormFields.siteRegistro)}
              onChange={formik.handleChange(FormFields.siteRegistro)}
              error={
                !!(formik.touched.siteRegistro && formik.errors.siteRegistro)
              }
              errorText={formik.errors[FormFields.siteRegistro]}
            />
            <Tooltip
              title="On what FTX website did the client registered?"
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
          <GridSyled>
            <RadioButtonSyled>
              <Typography variant="body1" color="textPrimary" paddingBottom={1}>
                O cliente mantinha uma conta digital com a FTX antes de
                11/11/2022?
              </Typography>
              <RadioGroupStyled
                name={FormFields.contaAnteriorFalencia}
                value={String(formik.values.contaAnteriorFalencia)}
                onChange={(event) => {
                  const value = event.target.value === 'true';
                  formik.setFieldValue(FormFields.contaAnteriorFalencia, value);
                }}
              >
                <FormControlLabel
                  value="true"
                  control={<Radio />}
                  label="Sim"
                />
                <FormControlLabel
                  value="false"
                  control={<Radio />}
                  label="Não"
                />
              </RadioGroupStyled>
            </RadioButtonSyled>
            <Tooltip
              title="Did the client maintain a diigital wallet with FTX before 11/11/2022?"
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
          <GridSyled>
            <TextFieldStyled
              variant="outlined"
              size="medium"
              label="Que tipo de conta o cliente se registrou na FTX?"
              value={formik.values.tipoConta}
              onBlur={formik.handleBlur(FormFields.tipoConta)}
              onChange={formik.handleChange(FormFields.tipoConta)}
              error={!!(formik.touched.tipoConta && formik.errors.tipoConta)}
              errorText={formik.errors[FormFields.tipoConta]}
            />
            <Tooltip
              title="What type of account did the client register for on FTX?"
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
          <GridSyled>
            <TextFieldStyled
              variant="outlined"
              size="medium"
              label="Que tipos de ativos foram usados para abrir a conta?"
              value={formik.values.tipoAtivoInvestido}
              onBlur={formik.handleBlur(FormFields.tipoAtivoInvestido)}
              onChange={formik.handleChange(FormFields.tipoAtivoInvestido)}
              error={
                !!(
                  formik.touched.tipoAtivoInvestido &&
                  formik.errors.tipoAtivoInvestido
                )
              }
              errorText={formik.errors[FormFields.tipoAtivoInvestido]}
            />
            <Tooltip
              title="What type of assets were used to open the account?"
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
          <GridSyled>
            <TextFieldStyled
              variant="outlined"
              size="medium"
              label="Qual era o valor aproximado dos depósitos na conta em 11/11/2022?"
              value={formik.values.valorDepositos}
              onBlur={formik.handleBlur(FormFields.valorDepositos)}
              onChange={formik.handleChange(FormFields.valorDepositos)}
              error={
                !!(
                  formik.touched.valorDepositos && formik.errors.valorDepositos
                )
              }
              errorText={formik.errors[FormFields.valorDepositos]}
            />
            <Tooltip
              title="What was the approximate value of  total deposits in the account on 11/11/2022?"
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
          <GridSyled>
            <TextFieldStyled
              variant="outlined"
              size="medium"
              label="Por favor, forneça qualquer outra informação adicional para verificar o pedido."
              value={formik.values.observacao}
              onBlur={formik.handleBlur(FormFields.observacao)}
              onChange={formik.handleChange(FormFields.observacao)}
              error={!!(formik.touched.observacao && formik.errors.observacao)}
              errorText={formik.errors[FormFields.observacao]}
            />
            <Tooltip
              title="Please provide any other supporting information to verify the claim."
              arrow
            >
              <HelpOutlineIcon />
            </Tooltip>
          </GridSyled>
        </FormStyled>
        <ButtonGridSyled>
          <Button
            onClick={formik.submitForm}
            size="large"
            variant="contained"
            color="primary"
            disabled={!formik.dirty}
          >
            Salvar informações
          </Button>
        </ButtonGridSyled>
      </DialogContent>
    </Dialog>
  );
}
