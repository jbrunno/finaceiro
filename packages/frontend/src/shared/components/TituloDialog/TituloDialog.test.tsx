import { render } from '@test-utils/component-testing';
import { TituloDialog } from './TituloDialog';
import { tituloQuestionario } from './TituloDialog.mock';

test('should render TituloDialog', () => {
  const { container } = render(
    <TituloDialog
      open
      onClose={jest.fn()}
      onSaved={jest.fn()}
      tituloQuestionario={tituloQuestionario}
    />,
  );
  expect(container).toMatchSnapshot();
});
