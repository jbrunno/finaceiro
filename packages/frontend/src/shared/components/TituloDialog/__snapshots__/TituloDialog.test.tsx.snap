// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TituloDialog 1`] = `
<div
  aria-hidden="true"
/>
`;
